src/dot.ts

```diff
@@ -125,20 +135,24 @@ export class Dot extends Modifier {
     return this;
   }
 
-  setDotShiftY(y) {
+  setDotShiftY(y: number): this {
     this.dot_shiftY = y;
     return this;
   }
 
-  draw() {
-    this.checkContext();
+  draw(): void {
+    const ctx = this.checkContext();
     this.setRendered();
+    if (!this.note || !this.note.getStave() || this.index === undefined) {
+      throw new Vex.RERR('NoStaveNoteIndex', 'Drawing a dot requires a stave, a note, and an index.');
+    }
 
-    if (!this.note || this.index === null) {
-      throw new Vex.RERR('NoAttachedNote', "Can't draw dot without a note and index.");
+    const stave = this.note.getStave();
```

Comment:
We're already checking for stave and throw an exception above. Maybe:

```typescript
draw() : void {
  const ctx = ...;
  const stave = this.note.getStave();
  if (!stave || !this.note !! this.index === undefined) {
     throw ...
  }
```

  this.setRendered()

Reply:
Good catch! But we cannot place `this.note.getStave()` before checking the for the existence of `this.note`
I have removed the duplication anyway, thanks.
Done